### PR TITLE
Upgrade to 4.3 version of Remoting.

### PR DESCRIPTION
**Pull request by jeffret-b**
This version of Remoting provides a fix to the WebSocket implementation for large payloads. See [JENKINS-61409](https://issues.jenkins-ci.org/browse/JENKINS-61409). This Remoting upgrade needs to be paired with #4596 to complete the master side of the correction. This change is breaking in terms of WebSockets but as this feature is still considered Beta, this is not considered a breaking change for the agent or server generally.

This Remoting release also includes minor maintenance improvements, mostly in tests.

See the [Remoting changelog](https://github.com/jenkinsci/remoting/releases/tag/remoting-4.3)
### Proposed changelog entries

* Upgrade to Remoting 4.3 to fix an issue with large payloads over WebSockets. Requires an agent upgrade for the Beta WebSocket feature. (issue 61409)

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
Remoting library update. Tests have been added in the related PRs.
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -101,7 +101,7 @@ THE SOFTWARE.
     <maven-war-plugin.version>3.2.3</maven-war-plugin.version>
 
     <!-- Bundled Remoting version -->
-    <remoting.version>4.3-20200323.083513-2</remoting.version><!-- https://github.com/jenkinsci/remoting/pull/373 -->
```
Comment (jglick) [inline pom.xml]: You forgot to merge with #4596; did that now.

**Review by jglick — APPROVED** [review]

**Comment by jglick** [comment]
Note that for LTS (2.222.2) we will want to have a backport of this fix to Remoting.

File: pom.xml (hunk 1 above)
Comment (jeffret-b) [reply pom.xml]: Thanks for catching that.

**Review by res0nance — APPROVED** [review]
LGTM

**Comment by res0nance** [comment]
Retriggering CI

**Comment by res0nance** [comment]
There seems to be a weird failure where an XML fails to read. I'm not sure what it all means

**Review by Vlatombe — APPROVED** [review]

**Pull request merged**